proto/storage/collector_runtime_config.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+
+package storage;
+
+import "google/protobuf/timestamp.proto";
+import "storage/resource_collection.proto";
+import "storage/user.proto";
+
+option go_package = "storage";
+option java_package = "io.stackrox.proto.storage";
+
+message CollectorRuntimeConfigACK {
+  string error = 1;
+  Action action = 2;
+
+  enum Action {
+    ACK = 0;
+    NACK = 1;
+  }
+}
+
+enum RuntimeFeature {
+  PROCESSES = 0;
+  NETWORK_CONNECTIONS = 1;
+  NETWORK_ENDPOINTS = 2;
+}
+
+message Status {
+  message ProcessStatus {
+    bool enabled = 1;
+  }
+
+  message NetworkConnectionStatus {
+    bool enabled = 1;
+    bool aggregate_external = 2;
+  }
+
+  message NetworkEndpointStatus {
+    bool enabled = 1;
+    bool processes_listening_on_port = 2;
+  }
+
+  oneof status {
+    ProcessStatus process_status = 1;
+    NetworkConnectionStatus network_connection_status = 2;
+    NetworkEndpointStatus network_endpoint_status = 3;
+  }
+}
+
+message RuntimeFeatureConfig {
+  message RuntimeRule {
+    string resource_collection_name = 1;
+    Status status = 2;
+    google.protobuf.Timestamp created_at = 3;
+    google.protobuf.Timestamp last_updated = 4;
+    SlimUser created_by = 5;
+    SlimUser updated_by = 6;
+  }
+
+  RuntimeFeature feature = 1;
+  Status default_status = 2;
+  repeated RuntimeRule rules = 3;
+  google.protobuf.Timestamp created_at = 4;
+  google.protobuf.Timestamp last_updated = 5;
+  // The default is not created by a user, but by central so there is no created_by.
```

Comment:
If you wanted you could still have a created by.  So the default would be created by `SYSTEM` or something along those lines.  I'm not sure if later there would ever be a need for the user to create these from scratch or not.

Reply:
It doesn't look like there is a user called `SYSTEM` in the protobuf definitions. I don't know what the implications of creating that would be. I think it is probably safest not to have a `created_by` here.

Reply:
No there wouldn't be.  You would either have to add an enum as such or use a constant when first creating this thing.  I was just pointing it out as an option.  If it doesn't make sense then there is no reason to do it.

Reply:
I have decided not to include the user information in the internal API. It will still be in the user facing API and table definitions.